core/src/main/scripts/import_data_validator/validateData.py

```diff
@@ -1422,13 +1427,139 @@ class SegValidator(Validator):
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
+                    # not a valid coordinate usable in further validations
+                    del parsed_coords[col_name]
+            elif col_name == 'num.mark':
+                if not self.checkInt(value):
+                    self.logger.error(
+                        'Number of probes is not an integer',
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
+            elif col_name == 'seg.mean':
+                if not self.checkFloat(value):
+                    self.logger.error(
+                        'Mean segment copy number is not a number',
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
+            else:
+                raise RuntimeError('Could not validate column type: ' +
+                                   col_name)
+
+        if (
+                'loc.start' in parsed_coords and 'loc.end' in parsed_coords and
+                parsed_coords['loc.start'] >= parsed_coords['loc.end'] - 1):
+            self.logger.error(
+                'Start position of segment is not lower than end position',
+                extra={'line_number': self.line_number,
+                       'cause': '{}/{}'.format(parsed_coords['loc.start'],
+                                               parsed_coords['loc.end'])})
+
+        # TODO check for overlap and low genome coverage
+        # this could be implemented by sorting the segments for a patient
+        # by (chromosome and) start position and checking if the start position
+        # of each segment comes after the end position of the previous one,
+        # meanwhile adding up the number of (non-overlapping) bases covered on
+        # that chromosome in that patient.
+
+    @staticmethod
+    def load_chromosome_lengths(genome_build):
+
+        """Get the length of each chromosome from USCS and return a dict.
+
+        The dict will not include unplaced contigs, alternative haplotypes or
+        the mitochondrial chromosome.
+        """
+
+        chrom_size_dict = {}
+        chrom_size_url = (
+            'http://hgdownload.cse.ucsc.edu'
+            '/goldenPath/{build}/bigZips/{build}.chrom.sizes').format(
+                build=genome_build)
+        chrom_size_file = urllib2.urlopen(chrom_size_url)
+
+        for line in chrom_size_file:
+            try:
+                # skip comment lines
+                if line.startswith('#'):
+                    continue
+                cols = line.split('\t', 1)
+                if not (len(cols) == 2 and
+                        cols[0].startswith('chr')):
+                    raise IOError()
```

Comment:
seems like a bad practice to me. I would rather put the message in this IOError. Later in your except part you can then give the error "Unexpected response...." + the error you give here.